src/TNBase.DataStorage/Service/PreferencesService.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TNBase.Objects;
+
+namespace TNBase.DataStorage.Service
+{
+    public class PreferencesService
+    {
+        private readonly ITNBaseContext context;
+        private readonly ApplicationSettingsBase applicationSettings;
+
+        public PreferencesService(ITNBaseContext context, ApplicationSettingsBase applicationSettings)
+        {
+            this.context = context;
+            this.applicationSettings = applicationSettings;
+        }
+
+        public Preferences GetPreferences()
+        {
+            Preferences result;
+
+            result = context.Preferences.FirstOrDefault();
```

Comment:
I'd like to suggest a slightly different approach if that's OK. Your current setup is not very scalable as for any additional preference value we would require an additional SQL column. This could get out of hand pretty quickly. An alternative would be to create a table with Key and Value columns where each new preference would just be another record in the database. Then, we can have GetPreference method to accept either a string or enum as a Key. This way, service will not have to change every time we need another setting.

Thinking about it, my previous comment about options probably would have to be changed even further to have a dictionary of DefaultPreferences so you could do something like this:
```suggestion
            result = context.Preferences.FirstOrDefault(x=>x.Key == key) ?? options.DefaultPreferences[key];
```
Hope that's not too much refactoring. What do you think?